caput/pipeline/runner/_core.py

```diff
@@ -44,20 +79,31 @@ def run_pipeline(configfile, profile, profiler, mpi_abort, psutil):
         Manager.from_yaml_file(configfile, psutil_profiling=psutil).run()
 
 
-def template_run(templatefile, var, *args, **kwargs):
-    """Run a pipeline from the given `templatefile`.
+def template_run(templatefile: str, var: Any, *args: Any, **kwargs: Any):  # noqa: D417
+    r"""Run a pipeline from the given `templatefile`.
 
     Template variable substitutions are specified with `--var <varname>=<val>`
     arguments, with one for each variable. `<val>` may be a comma separated list, in
     which case item represents a separate value that is processed. Values *must* not
     contain a comma themselves. If multiple variables are specified, each with multiple
     substitutions the outer product of all possible values is generated.
+
+    Parameters
+    ----------
+    templatefile : str
+        A yaml file containing the pipeline template.
+    var : Any
+        A list of variable substitutions to apply to the template.
+    \*args : Any
+        Positional arguments to pass to `run_pipeline`.
+    \**kwargs : Any
+        Keyword arguments to pass to `run_pipeline`.
     """
     for tfh_name in _from_template(templatefile, var):
         run_pipeline(tfh_name, *args, **kwargs)
 
 
-def _from_template(templatefile, var):
+def _from_template(templatefile: str, var: Any) -> Generator[str]:
```

Comment:
I don't think `templatefile` can be a `str`: A `.read()` call is done on it down on line 128.

(If this type needs to be changed, the change will then need to percolate up to all callers of `_from_template`.)

Reply:
Ah - I think it comes from the `cli` click type. I'm going to change that, but not in this PR.